example/windows/main.cpp

```diff
@@ -84,9 +85,30 @@ int APIENTRY wWinMain(HINSTANCE instance, HINSTANCE prev, wchar_t *command_line,
   // Parent and resize Flutter view into top-level window.
   window.SetChildContent(flutter_controller.GetNativeWindow());
 
-  // Run messageloop with a hook for flutter_view to do work.
-  window.RunMessageLoop(
-      [&flutter_controller]() { flutter_controller.ProcessMessages(); });
+  // Run messageloop with a hook for flutter_controller to do work until
+  // the window is closed.
+  std::chrono::nanoseconds wait_duration(0);
+  // Run until the window is closed.
+  while (window.GetHandle() != nullptr) {
+    MsgWaitForMultipleObjects(0, NULL, FALSE,
+                              static_cast<DWORD>(wait_duration.count() / 1000),
+                              QS_ALLINPUT);
+    MSG message;
+    // All pending Windows messages must be processed; MsgWaitForMultipleObjects
+    // won't return again for items left in the queue after PeekMessage.
+    while (PeekMessage(&message, nullptr, 0, 0, PM_REMOVE)) {
+      if (message.message == WM_QUIT) {
+        window.Destroy();
+        break;
+      }
+      TranslateMessage(&message);
+      DispatchMessage(&message);
+    }
+    // Allow Flutter to process its messages.
+    // TODO: Consider interleaving processing on a per-message basis to avoid
```

Comment:
Should we file a bug on this?

Reply:
I figured we'd let it bake in the real world for a while to see if it's something that actually comes up in practice. The code gets uglier if we do this, which is why I held off for now.